microsoft/PowerToys
# Detect if all Drives are indexed and show a warning if they aren't

## Summary of the Pull Request
* This PR looks identifies if the enhanced Indexing mode is turned on in the users system, if so, it does not return any results from the indexer plugin but let's the user know about it. This warning can be disabled from the settings page of PowerToys run.

 
## References
- Finding out if enhanced indexing mode is turned on - https://www.tenforums.com/tutorials/120447-turn-off-enhanced-mode-search-indexer-windows-10-a.html.


## PR Checklist
* [x] Applies to #2041


## Detailed Description of the Pull Request / Additional comments
- A `RegistryWrapper` class has been added which  obtains the Local machine registry key to identify if all drives are indexed using the Registry Entry `EnableFindMyFiles`.
- If the enhanced indexing mode is off, then the following warning is displayed.
![image](https://user-images.githubusercontent.com/28739210/87596842-a4805a00-c6a5-11ea-8b5b-ec5ddefe6c5e.png)
- If users decide to not enable the enhanced mode of indexing they can disable it in the PowerToys settings.
![image](https://user-images.githubusercontent.com/28739210/87596888-bb26b100-c6a5-11ea-805d-b7345b169fb0.png)

Gif showing user flow - 
![image](https://user-images.githubusercontent.com/28739210/87565598-b39be380-c676-11ea-9edc-ee358c81ae46.gif)


## Validation Steps Performed
- Added tests for the PowerToys settings to ensure that the PT Run indexer settings are overridden when the checkbox is selected/deselected.
- Added tests to the indexer plugin to ensure that it works as expected
- Manually validated that the settings are stored in the json file and carry over the next time we open powertoys. 
- Tested that it works as expected when the json file does not have an entry for this checkbox.
